### DISCUSSION TITLE
Latest builds unstable?

### DISCUSSION THREAD

**Discussion by steve6375** [a1ive/grub2-filemanager#326] [General]
I compiled todays version and am experiencing problems of display corruption, hanging, black screen, etc. after  browsing around and using F1-F6, selecting folders which contain many files, etc.
Noticed this for several of the most recent versions.
I have no \boot\grubfm\config file and it still does it.
Looks like some sort of memory leak issue or maybe bad syntax (if - fi or { - } unbalanced?).
You have to muck about quite a lot selecting different menu folders, etc. but eventually it crashes.
Tested on real system and VBox5 - same results.
It is hard to pinpoint the issue - just seems to happen if you use the menu system a lot.
Also happens if I used text mode instead of a theme.

e.g. in text mode - stuck here - will not move up and no F1-F6 keys.
![image](https://user-images.githubusercontent.com/1469350/103479111-436b5080-4dc3-11eb-94f2-444259b2464e.png)
at other times it may crash\hang.

**Comment by a1ive** [comment]
https://github.com/a1ive/grub/issues/61
This is not caused by a memory leak.
The previous version had no problems instead because of memory leaks.
Now the memory leak is solved (https://github.com/a1ive/grub/commit/34c9f905f9b559544af7ab441fb23dd2abee10fe), but dangling pointers appear in some places.
grub2 is not as reliable as we thought. It's is too complicated, so I may abandon grub2 in the future and write another bootloader from scratch.

**Comment by a1ive** [comment]
revert this commit https://github.com/a1ive/grub/commit/34c9f905f9b559544af7ab441fb23dd2abee10fe will fix this issue, but it will cause memory leak.

**Comment by steve6375** [comment]
![image](https://user-images.githubusercontent.com/1469350/103479806-19685d00-4dc8-11eb-84ec-1444a460c5cf.png)
Maybe better to have memory leak than crashing?
